src/Servers/Kestrel/samples/http2cat/http2cat.csproj

```diff
@@ -22,6 +22,7 @@
     <Reference Include="Microsoft.AspNetCore.Server.Kestrel.Core" />
     <Reference Include="Microsoft.Extensions.Hosting" />
     <Reference Include="Microsoft.Extensions.Logging.Console" />
+    <Reference Include="Microsoft.AspNetCore.Testing" />
```

Comment:
Hmmm, I'm not sure about this.

I think the timeout extensions file should be in a shared location and http2cat and Microsoft.AspNetCore.Testing each reference it.

An `#if` can then be used to control whether the type is public or internal. http2cat's usage would be internal.

Reply:
Yes, that was my plan as well, however I missed this trick with ```#if```` on the visibility. I think that would solve the problem where a project references both Microsoft.AspNetCore.Testing and Http2cat too. Not sure if it solves all the issues where instead of http2cat project, only the files are referenced - but I will give it a go.
Question 1: Can I use a new symbol to use in the  ```#if```` compiler condition, or do you have a suggested one?

Question 2: what should be the namespace?
In this case a suggestion for ```System.Threading.Tasks``` seems reasonable, but it might give a false impression that these extension methods were designed to be used in shipping projects.
Or a second approach could be to ```#if``` else on the namespace too.

Reply:
http2cat says though: "Http2Cat is a low level Http2 testing framework designed to excersize a server with frame level control. This can be useful for unit testing and compat testing."
https://github.com/dotnet/aspnetcore/blob/main/src/Shared/Http2cat/Http2CatReadMe.md

Reply:
I have added a 2nd PR: https://github.com/dotnet/aspnetcore/pull/31768 as a comparison. After implementing the suggested changes, I found this solution cleaner, hence the separate PR. If that is the preferred, we can continue with that one.

Reply:
@halter73 Thoughts on http2cat referencing Microsoft.AspNetCore.Testing?

Reply:
Do you have a suggestion? :)

Reply:
> @halter73 Thoughts on http2cat referencing Microsoft.AspNetCore.Testing?

That should be fine, h2cat is only used in sample and test code. Did it cause any build problems?

Reply:
If not referenced it would have required some ```#if```-s, but no problems that could not be solved.

Reply:
I'm okay with this since it's easier. We don't ship it as a package or anything.